libraries/SITL/SIM_Frame.cpp

```diff
@@ -341,12 +350,69 @@ void Frame::load_frame_params(const char *model_json)
     if (fname == nullptr) {
         AP_HAL::panic("%s failed to load\n", model_json);
     }
-    picojson::value *obj = (picojson::value *)load_json(model_json);
-    if (obj == nullptr) {
-        AP_HAL::panic("%s failed to load\n", model_json);
+
+    jsmntok_t t[50];
```

Comment:
it should be good to have a simple load_json() method which stops us having to repeat all this logic in multiple places. I presume a wrapper like that, like we had with load_json() would be doable?